components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -1112,28 +1115,97 @@ void RewardsServiceImpl::OnURLLoaderComplete(
 
   if (response_body && !response_body->empty() && loader->ResponseInfo() &&
       base::Contains(loader->ResponseInfo()->mime_type, "json")) {
-    return data_decoder::JsonSanitizer::Sanitize(
-        *response_body,
-        base::BindOnce(
-            [](ledger::client::LoadURLCallback callback,
-               ledger::mojom::UrlResponse response,
-               data_decoder::JsonSanitizer::Result result) {
-              if (result.value) {
-                response.body = std::move(*result.value);
-              } else {
-                response.body = {};
-                VLOG(0) << "Response sanitization error: "
-                        << (result.error ? *result.error : "unknown");
-              }
-
-              std::move(callback).Run(response);
-            },
-            std::move(callback), std::move(response)));
+    return OnURLLoaderCompleteSanitizeJson(
+        std::move(callback), std::move(response), std::move(response_body));
   }
 
   std::move(callback).Run(response);
 }
 
+void RewardsServiceImpl::OnURLLoaderCompleteSanitizeJson(
+    ledger::client::LoadURLCallback callback,
+    ledger::mojom::UrlResponse response,
+    std::unique_ptr<std::string> response_body) {
+// On Android |RewardsServiceImpl::OnURLLoaderComplete| is executed in UI
+// thread. We have evidences of ANRs happened at |JsonSanitizer::Sanitize|.
+// By this reason sanitizing on Android is moved to non-UI thread, callback
+// is still invoked in UI thread.
+#if BUILDFLAG(IS_ANDROID)
+  OnURLLoaderCompleteSanitizeJsonAndroid(
+      std::move(callback), std::move(response), std::move(response_body));
+#else
+  OnURLLoaderCompleteSanitizeJsonDesktop(
+      std::move(callback), std::move(response), std::move(response_body));
+#endif  // BUILDFLAG(IS_ANDROID)
+}
+
+#if BUILDFLAG(IS_ANDROID)
+
+void RewardsServiceImpl::OnURLLoaderCompleteSanitizeJsonAndroid(
+    ledger::client::LoadURLCallback callback,
+    ledger::mojom::UrlResponse response,
+    std::unique_ptr<std::string> response_body) {
+  json_sanitizer_task_runner_->PostTask(
+      FROM_HERE,
+      base::BindOnce(
+          [](std::unique_ptr<std::string> response_body,
+             ledger::client::LoadURLCallback callback,
+             ledger::mojom::UrlResponse response,
+             const scoped_refptr<base::SequencedTaskRunner>
+                 post_response_runner) {
+            data_decoder::JsonSanitizer::Sanitize(
+                *response_body,
+                base::BindOnce(
+                    [](ledger::client::LoadURLCallback callback,
+                       ledger::mojom::UrlResponse response,
+                       const scoped_refptr<base::SequencedTaskRunner>
```

Comment:
This could be reference-to-`const` instead, so that we don't incur the cost of an additional move constructor (which kicks in when the argument gets passed to the callback as an rvalue).

Reply:
thanks, it is now reference-to-const at 
https://github.com/brave/brave-core/pull/17354/commits/993eaaa906128939670c9b12a03076dca14879ab#diff-b3817ac19a1c6ad3db2832fc878795ccc8a6f5fc0ae281cdf70d85a3a0a3d324R1147